core/service/src/client/mod.rs

```diff
@@ -6040,6 +6040,310 @@ pub(crate) mod tests {
+        // because every party will load all the keys and each ServerKey is 1.5 GB
+        // and each private key share is 1 GB. Using 7 parties fails on a 32 GB machine.
+
+        let amount_parties = 4;
+        let param = FheParameter::Default;
+        let dkg_param: WrappedDKGParams = param.into();
+
+        let key_id_1: RequestId = derive_request_id(&format!(
+            "default_insecure_dkg_backup_1_{amount_parties}_{param:?}",
+        ))
+        .unwrap();
+        let key_id_2: RequestId = derive_request_id(&format!(
+            "default_insecure_dkg_backup_2_{amount_parties}_{param:?}",
+        ))
+        .unwrap();
+
+        let test_path = None;
+        purge(test_path, test_path, test_path, &key_id_1, amount_parties).await;
+        purge(test_path, test_path, test_path, &key_id_2, amount_parties).await;
+        let (_kms_servers, kms_clients, internal_client) =
+            threshold_handles(*dkg_param, amount_parties, true, None, None).await;
+
+        // Purge the backup after booting the server to ensure nothing is backeup automatically during boot
+        purge_backup(test_path, amount_parties).await;
+        let keys_1 = run_threshold_keygen(
+            param,
+            &kms_clients,
+            &internal_client,
+            None,
+            &key_id_1,
+            None,
+            None,
+            true,
+        )
+        .await;
+        // check that we have the new mod switch key
+        let (client_key_1, _, server_key_1) = keys_1.clone().get_standard();
+        check_conformance(server_key_1, client_key_1);
+        let panic_res = std::panic::catch_unwind(|| keys_1.get_decompression_only());
+        assert!(panic_res.is_err());
+
+        let keys_2 = run_threshold_keygen(
+            param,
+            &kms_clients,
+            &internal_client,
+            None,
+            &key_id_2,
+            None,
+            None,
+            true,
+        )
+        .await;
+
+        // check that we have the new mod switch key
+        let (client_key_2, _, server_key_2) = keys_2.clone().get_standard();
+        check_conformance(server_key_2, client_key_2);
+        let panic_res = std::panic::catch_unwind(|| keys_2.get_decompression_only());
+        assert!(panic_res.is_err());
+
+        // Generated key, delete private storage
+        for i in 1..=amount_parties {
+            let mut priv_storage: FileStorage = FileStorage::new(
+                test_path,
+                StorageType::PRIV,
+                Some(Role::indexed_from_one(i)),
+            )
+            .unwrap();
+            delete_all_at_request_id(&mut priv_storage, &key_id_1).await;
+            delete_all_at_request_id(&mut priv_storage, &key_id_2).await;
+        }
+        // Make sure file is actually deleted
+        tokio::time::sleep(tokio::time::Duration::from_millis(300)).await;
+        // Now try to restore both keys
+        let mut resp_tasks = JoinSet::new();
```

Comment:
before doing this, might be useful to try and do a decryption, which should fail

Reply:
It would not work. Restore only adds the keys to the storage, it does not reload them. A reboot is currently needed. Given that we likely want to remove this method once the proper backup is in place, I chose not to spend time adding this feature and just documented in the readme that a reboot is needed.